pandera/backends/polars/components.py

```diff
@@ -87,7 +87,7 @@ def check_dtype(
 
         if schema.dtype is not None:
             obj_dtype = check_obj.schema[schema.name]
-            passed = obj_dtype is schema.dtype
+            passed = obj_dtype.is_(schema.dtype)
```

Comment:
Because some dtypes objects are in instantiated in the schema, for instance `Duration(time_unit="us")`, the `is` comparison doesn't work here. So I used the `is_` method provided by the Polars `DataType` API instead